src/main/resources/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval/render-classpaths.js

```diff
@@ -0,0 +1,152 @@
+
+            document.getElementById('pendingClasspathEntries').appendChild(block);
+        });
+        document.getElementById('pendingClasspathEntries').style.display = '';
+    }
+}
+
+function renderApprovedClasspathEntries(approvedClasspathEntries) {
+    if (approvedClasspathEntries.length == 0) {
+        document.getElementById('approvedClasspathEntries-none').style.display = '';
+        Array.from(document.getElementById('approvedClasspathEntries').children).forEach(function(e){e.remove()});
+        document.getElementById('approvedClasspathEntries').style.display = 'none';
+        document.getElementById('approvedClasspathEntries-clear').style.display = 'none';
+    } else {
+        document.getElementById('approvedClasspathEntries-none').style.display = 'none';
+        Array.from(document.getElementById('approvedClasspathEntries').children).forEach(function(e){e.remove()});
+        approvedClasspathEntries.forEach(function(e) {
+            var block = document.createElement('p');
+            block.setAttribute('id', 'acp-' + e.hash);
+            var deleteButton = document.createElement('button');
+            deleteButton.setAttribute('class', 'delete');
+            deleteButton.setAttribute('hash', e.hash);
+            deleteButton.textContent = 'Delete';
+            deleteButton.addEventListener('click', function() {
+                if (confirm('Really delete this approved classpath entry? Any existing scripts using it will need to be rerun and the entry reapproved.')) {
+                    denyApprovedClasspathEntry(this.getAttribute('hash'));
+                }
+            });
+            block.appendChild(deleteButton);
+            var code = document.createElement('code');
+            code.setAttribute('title', e.hash);
+            code.textContent = e.path;
+            block.appendChild(code);
+
+            document.getElementById('approvedClasspathEntries').appendChild(block);
+        });
+        document.getElementById('approvedClasspathEntries').style.display = '';
+        document.getElementById('approvedClasspathEntries-clear').style.display = '';
+    }
+}
+
+function renderClasspaths(r) {
+    renderPendingClasspathEntries(r.responseObject()[0]);
+    renderApprovedClasspathEntries(r.responseObject()[1]);
+}
+
+function approveClasspathEntry(hash) {
+    mgr.approveClasspathEntry(hash, function(r) {
+        renderClasspaths(r);
+    });
+}
+function denyClasspathEntry(hash) {
+    mgr.denyClasspathEntry(hash, function(r) {
+        renderClasspaths(r);
+    });
+}
+function denyApprovedClasspathEntry(hash) {
+    mgr.denyApprovedClasspathEntry(hash, function(r) {
+        renderClasspaths(r);
+    });
+}
+function clearApprovedClasspathEntries() {
+    mgr.clearApprovedClasspathEntries(function(r) {
+        renderClasspaths(r);
+    });
+}
+
+document.addEventListener('DOMContentLoaded', function() {
+    mgr.getClasspathRenderInfo(function(r) {
+        renderClasspaths(r);
+    });
+});
```

Comment:
Diff from original code:

```js
@@ -1,4 +1,3 @@
-var mgr = <st:bind value="${it}"/>;
 function hideScript(hash) {
     document.getElementById('ps-' + hash).remove();
 }
@@ -56,14 +55,6 @@
     } else {
         document.getElementById('pendingClasspathEntries-none').style.display = 'none';
         Array.from(document.getElementById('pendingClasspathEntries').children).forEach(function(e){e.remove()});
-        /*
-           Create a list like:
-            <p id="pcp-${pcp.hash}">
-                <button class="approve" onclick="approveClasspathEntry('${pcp.hash}')">Approve</button> /
-                <button class="deny"    onclick="denyClasspathEntry('${pcp.hash}')">Deny</button> 
-                ${pcp.hash} (${pcp.path})
-            </p>
-         */
         pendingClasspathEntries.forEach(function(e) {
             var block = document.createElement('p');
             block.setAttribute('id', 'pcp-' + e.hash);
@@ -103,13 +94,6 @@
     } else {
         document.getElementById('approvedClasspathEntries-none').style.display = 'none';
         Array.from(document.getElementById('approvedClasspathEntries').children).forEach(function(e){e.remove()});
-        /*
-           Create a list like:
-            <p id="acp-${acp.hash}">
-                <button class="delete" onclick="denyApprovedClasspathEntry('${pcp.hash}')">Delete</button> 
-                ${acp.hash} (${acp.path})
-            </p>
-         */
         approvedClasspathEntries.forEach(function(e) {
             var block = document.createElement('p');
             block.setAttribute('id', 'acp-' + e.hash);
@@ -161,7 +145,7 @@
     });
 }
 
-window.addEventListener("load", function(){
+document.addEventListener('DOMContentLoaded', function() {
     mgr.getClasspathRenderInfo(function(r) {
         renderClasspaths(r);
     });
```

Reply:
Explanation of changes:

- `var mgr = <st:bind value="${it}"/>;` is replaced with CSP-compliant `<st:bind value="${it}" var="mgr"/>`
- Inline event handler definitions are not CSP compliant, but they were only in comments, so I simply deleted the comments.
- Used `DOMContentLoaded` as suggested in https://www.jenkins.io/doc/developer/security/csp/#inline-event-handlers